person.rb

```diff
@@ -0,0 +1,39 @@
+require_relative 'capitalize_decorator'
+require_relative 'trimmer_decorator'
+require_relative 'nameable'
+
+# for person
+class Person < Nameable
+  attr_reader :id
+  attr_accessor :name, :age
+
+  def initialize(id, age, name: 'Unknown', parent_permission: true)
+    super()
+    @id = id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def correct_name
+    @name
+  end
+end
+
+person1 = Person.new(1, 17, name: 'Abutalha', parent_permission: false)
```

Comment:
- [Optional] When creating a teacher or a student(person) the `id` parameter won't be required. It is advisable to generate random numbers for the id.